### PR TITLE
Reversed X-XSS

### PR DESCRIPTION
**Pull request by AaronSadlerUK**
### Prerequisites

Resolves# #13341

### Description
This reverses the health check to warn if it does exist with a link to the Mozilla website which explains why it should not exist

I have also changed the readmore button to be nullable (it's not always required) and added a bool to reverse the check in the base class

### CODE REVIEW

File: src/Umbraco.Core/EmbeddedResources/Lang/en.xml [hunk 1]
```diff
@@ -2358,9 +2358,13 @@ To manage your website, simply open the Umbraco backoffice and start adding cont
     <key alias="hSTSCheckHeaderNotFoundOnLocalhost">
       <![CDATA[The header <strong>Strict-Transport-Security</strong> was not found. This header should not be present on localhost.]]>
     </key>
-    <key alias="xssProtectionCheckHeaderFound"><![CDATA[The header <strong>X-XSS-Protection</strong> was found.]]></key>
+    <key alias="xssProtectionCheckHeaderFound">
+      <![CDATA[The header <strong>X-XSS-Protection</strong> was found. <strong>It is recommended not to add this header to your website</strong>.<br />
+    You can read about this on the <a href="https://docs.umbraco.com/umbraco-cms/extending/health-check" target="_blank" rel="noopener" class="btn-link -underline">Mozilla</a> website ]]>
```
Comment (tristanjthompson) [inline src/Umbraco.Core/EmbeddedResources/Lang/en.xml:2363]: Should this URL be https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection
```suggestion
    You can read about this on the <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection" target="_blank" rel="noopener" class="btn-link -underline">Mozilla</a> website ]]>
```

Comment (AaronSadlerUK) [reply src/Umbraco.Core/EmbeddedResources/Lang/en.xml]: Yes!

This is a case of brain working faster than fingers 😂

It's updated and pushed

**Comment by tristanjthompson** [comment]
Bristol Hackathon testing!

I've tested a vanilla install without the `X-XSS-Protection` header and the health check correctly passed and looks as follows ✔️ 
image


I've updated the install to produce the `X-XSS-Protection` header, and the health check correctly failed and looks as follows ✔️ 

image

Clicking the "Mozilla" link shold take me to the Mozilla info page with more info on the header, however it takes me to the Umbraco docs for the health check ❌ 
* Expected URL: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection
* Actual URL: https://docs.umbraco.com/umbraco-cms/extending/health-check

Just need to update that URL and I'm happy it's passing 👍

**Comment by AaronSadlerUK** [comment]
Corrected in commit https://github.com/umbraco/Umbraco-CMS/pull/13936/commits/5d07968cd3b4fe10a611e23dbdcb3a4a0d45840e

Thanks!

File: src/Umbraco.Core/EmbeddedResources/Lang/en_us.xml [hunk 2]
```diff
@@ -2461,7 +2461,9 @@ To manage your website, simply open the Umbraco backoffice and start adding cont
     <key alias="hSTSCheckHeaderNotFoundOnLocalhost">
       <![CDATA[The header <strong>Strict-Transport-Security</strong> was not found. This header should not be present on localhost.]]>
     </key>
-    <key alias="xssProtectionCheckHeaderFound"><![CDATA[The header <strong>X-XSS-Protection</strong> was found.]]></key>
+    <key alias="xssProtectionCheckHeaderFound">
+      <![CDATA[The header <strong>X-XSS-Protection</strong> was found. <strong>It is recommended not to add this header to your website</strong>.<br />
+    You can read about this on the <a href="https://docs.umbraco.com/umbraco-cms/extending/health-check" target="_blank" rel="noopener" class="btn-link -underline">Mozilla</a> website ]]></key>
```
Comment (tristanjthompson) [inline src/Umbraco.Core/EmbeddedResources/Lang/en_us.xml:2466]: Just been re-testing and it was still showing the old link!  Looks like it needs updating here too - d'oh!
```suggestion
    You can read about this on the <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection" target="_blank" rel="noopener" class="btn-link -underline">Mozilla</a> website ]]></key>
```

**Comment by nul800sebastiaan** [comment]
Teamwork, thanks very much both, this is now merged for 11.3! 👍

**Comment by nul800sebastiaan** [comment]
Hey @AaronSadlerUK! In all my excitement and GitHub not showing the builld failing I merged this with not a glance at the code, but unfortunately we have a few breaking changes there so we had to roll this one back unfortunately. It would be really great if you'd like to try this one again though!

The changes that were breaking: 

- Removing `public const XssProtectionCheck` is a binary breaking change
- Adding more arguments to the public class `BaseHttpHeaderCheck` is a breaking change
- And although making the ReadMoreLink property nullable isn't a binary breaking change, it is a compiler breaking change, since it might require adding a null check (and can even fail rebuilding when you have `<WarningsAsErrors>Nullable</WarningsAsErrors>` set).

I have't looked deeply into this PR at all but if you could address the above issues then we could give it another go! Thanks again for this attempt and sorry that I was a bit too quick on the trigger on merging this one.

**Comment by JasonElkin** [comment]
I don't think the `BaseHttpHeaderCheck` is a good abstraction for what we're trying to do here.

I started working on a PR to obsolete it. For a proper separation of concerns I think we need an "IHttpHeader" service (or helper) - it really doesn't make sense to encapsulate the general HTTP Header checks inside a healthcheck.

Also, judging by the MDN docs, we don't necessarily want to ensure it's not there - we need to check its actual value.

**Pull request merged**